spec/models/chatroom_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Chatroom, type: :model do
+  describe "Associations" do
+    it { should have_many(:chatroom_users) }
+    it { should have_many(:users).through(:chatroom_users) }
+    it { should have_many(:messages) }
+  end
+
+  describe "Validations" do
+    subject { FactoryBot.create(:chatroom) }
+
+    it { should validate_presence_of(:name) }
+    it { should validate_uniqueness_of(:name) }
+    it { should validate_length_of(:name), maximum: 50 }
+
+    it "is a valid within valid attributes" do
+      expect(subject).to be_valid
+    end
+
+    it "is not a valid within maximum length name more than 50" do
+      subject.name = Faker::Lorem.characters(number: 51)
+      expect(subject).to_not be_valid
+    end
+
+    it "is not a valid without name" do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+  end
+
+  describe "Scope" do
+    it "will show the chatroom with direct_message return false" do
+      expect(Chatroom.public_channels).to eq Chatroom.where(direct_message: false)
+    end
+
+    it "will show the chatroom with direct_message return true" do
+      expect(Chatroom.direct_messages).to eq Chatroom.where(direct_message: true)
+    end
+  end
+
+  describe "#search" do
+    it "return a chatroom with name like the key_search" do
+      expect(Chatroom.search("example").to_sql).to eq Chatroom.where("lower(chatrooms.name) LIKE '%example%'").distinct.to_sql
```

Comment:
kỳ vọng ra kết quả luôn anh, ví dụ như Chatroom.search("example").to_sql khi thực hiện thì ra là 'abc' thì to el 'abc'